src/newgrf.cpp

```diff
@@ -2604,6 +2604,18 @@ static ChangeInfoResult LoadTranslationTable(uint gvid, int numinfo, ByteReader
 	return CIR_SUCCESS;
 }
 
+/**
+ * Helper to read a DWord from the reader and to output it, with guaranteed '\0'-termination into a buffer.
+ * @param reader The source of the DWord.
+ * @param output The output buffer to write to.
+ */
+static void ReadDWordAsString(ByteReader *reader, char output[5])
+{
+	uint32 tempfix = reader->ReadDWord();
+	memcpy(output, &tempfix, 4);
```

Comment:
This would be wrong on big endian.
Do we support that?

Reply:
Good question; I just moved the code from below, so yeah... interesting question.